SpatialGDK/Source/Private/Interop/SnapshotManager.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SnapshotManager.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/GlobalStateManager.h"
+#include "Interop/SpatialReceiver.h"
+#include "SpatialConstants.h"
+#include "Utils/SchemaUtils.h"
+
+DEFINE_LOG_CATEGORY(LogSnapshotManager);
+
+using namespace improbable;
+
+void USnapshotManager::Init(USpatialNetDriver* InNetDriver)
+{
+	NetDriver = InNetDriver;
+	Receiver = InNetDriver->Receiver;
+	GlobalStateManager = InNetDriver->GlobalStateManager;
+}
+
+// WorldWipe will send out an expensive entity query for every entity in the deployment.
+// It does this by having an entity query for all entities that are not the GSM (workaround for not having the ability to make a query for all entities).
+// Once it has the response to this query, it will send deletion requests for all found entities and then one for the GSM itself.
+// Should only be triggered by the worker which is authoritative over the GSM.
+void USnapshotManager::WorldWipe(const USpatialNetDriver::PostWorldWipeDelegate& Delegate)
```

Comment:
`Delegate` -> `PostWorldWipeDelegate`? (I know it's the same as the delegate type, but the type is in a namespace)